src/charm.py

```diff
@@ -889,53 +849,47 @@ def _replication_password(self) -> str:
         """
         return self.get_secret("app", REPLICATION_PASSWORD_KEY)
 
-    def _install_apt_packages(self, _, packages: List[str]) -> None:
-        """Simple wrapper around 'apt-get install -y.
+    def _install_snap_packages(self, packages: List[str]) -> None:
+        """Installs package(s) to container.
 
-        Raises:
-            CalledProcessError if it fails to update the apt cache.
-            PackageNotFoundError if the package is not in the cache.
-            PackageError if the packages could not be installed.
+        Args:
+            packages: list of packages to install.
         """
-        try:
-            logger.debug("updating apt cache")
-            apt.update()
-        except subprocess.CalledProcessError as e:
-            logger.exception("failed to update apt cache, CalledProcessError", exc_info=e)
-            raise
-
-        for package in packages:
+        for snap_name, snap_channel in packages:
             try:
-                apt.add_package(package)
-                logger.debug(f"installed package: {package}")
-            except apt.PackageNotFoundError:
-                logger.error(f"package not found: {package}")
-                raise
-            except apt.PackageError:
-                logger.error(f"package error: {package}")
+                snap_cache = snap.SnapCache()
+                snap_package = snap_cache[snap_name]
+
+                snap_package.ensure(snap.SnapState.Latest, channel=snap_channel)
+
+            except (snap.SnapError, snap.SnapNotFoundError) as e:
+                logger.error(
+                    "An exception occurred when installing %s. Reason: %s", snap_name, str(e)
+                )
                 raise
 
-    def _install_pip_package(self, package: str, user: Optional[str] = None) -> None:
-        """Simple wrapper around pip install.
+    def _patch_snap_seccomp_profile(self) -> None:
+        """Patch snap seccomp profile to allow chmod on pgBackRest restore code.
```

Comment:
Sidestepping and [issue](https://github.com/pgbackrest/pgbackrest/issues/2036) with sandboxing pgbackrest